digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporterSpec.scala

```diff
@@ -375,6 +382,46 @@ class CsvDataImporterSpec extends AuthenticatedApiSpec with BeforeAndAfter {
         csvRow = trafficSignCsvImporter.rowToString(defaultValues ++ assetFields)))))
   }
 
+  test("validation for lanes import fails if parameters are missing", Tag("db")) {
+    def createBadCsvLanes(lanes: Map[String, Any]*): String = {
+      val headers = List("kaista")
+      val rows = lanes.map { lane =>
+        headers.map { key => lane.getOrElse(key, "") }.mkString(";")
+      }.mkString("\n")
+      headers.mkString(";") + "\n" + rows
+    }
+
+    val laneRow = Map("kaista" -> 11)
+
+    val invalidCsv = csvToInputStream(createBadCsvLanes(laneRow))
+    val assets = lanesCsvImporter.processing(invalidCsv, testUser)
+
+    assets.incompleteRows.size should be (1)
+    assets.incompleteRows.head.missingParameters should contain allOf ("katyyppi", "tie", "osa", "ajorata", "aet", "let")
+  }
+
+  test("validation for lanes import fails if parameters are malformed", Tag("db")) {
+    val laneRow1 = Map("kaista" -> 100, "katyyppi" -> 1, "tie" -> "abc", "osa" -> 67, "ajorata" -> 671, "aet" -> 0, "let" -> 2113)
+    val laneRow2 = Map("kaista" -> 13, "katyyppi" -> 100, "tie" -> 2, "osa" -> 67, "ajorata" -> 671, "aet" -> "", "let" -> "")
+
+    val invalidCsv = csvToInputStream(createCsvLanes(laneRow1, laneRow2))
+    val assets = lanesCsvImporter.processing(invalidCsv, testUser)
+
+    assets.malformedRows.size should be (2)
+    assets.malformedRows.last.malformedParameters should contain allOf ("kaista", "tie")
+    assets.malformedRows.head.malformedParameters should contain allOf ("katyyppi", "aet", "let")
+  }
+
+  test("validation for lanes import fails if parameters combinations are invalid", Tag("db")) {
+    val laneRow = Map("kaista" -> 12, "katyyppi" -> 1, "tie" -> 7, "osa" -> 67, "ajorata" -> 2, "aet" -> 0, "let" -> 1000)
+
+    val invalidCsv = csvToInputStream(createCsvLanes(laneRow))
+    val assets = lanesCsvImporter.processing(invalidCsv, testUser)
+
+    assets.notImportedData.size should be (1)
+    assets.notImportedData.head.csvRow should be (lanesCsvImporter.rowToString(laneRow))
+  }
+
```

Comment:
I think it's missing some test in case of success, I only see tests for fails and none with a correct creation.

Reply:
Done